yoyodyne/models/transducer.py

```diff
@@ -45,20 +45,21 @@ def __init__(
             *args: passed to superclass.
             **kwargs: passed to superclass.
         """
+        # Alternate outputs than dataset targets.
+        kwargs["output_size"] = len(expert.actions)
+        super().__init__(*args, **kwargs)
         # Model specific variables.
         self.expert = expert  # Oracle to train model.
         self.actions = self.expert.actions
         self.substitutions = self.actions.substitutions
         self.insertions = self.actions.insertions
-        # Alternate outputs than dataset targets.
-        kwargs["output_size"] = len(expert.actions)
-        super().__init__(*args, **kwargs)
         # Target embeddings use alternate padding.
         self.target_embeddings = self.init_embeddings(
             num_embeddings=self.output_size,
             embedding_size=self.embedding_size,
             pad_idx=self.actions.end_idx,
         )
+        self.loss_func = TransducerNoFeatures.loss_func
```

Comment:
This requires documentation. You're trying to say that you're going to overload it later? Then this should be no-op. Or are you trying to say something else?

Reply:
I'm having a hard time making this non-obtrusive. The transducer forward pass computes a loss with it, so its loss function is just a dummy wrapper for referencing a tuple. Current inheritance order wants to call `base`'s loss function finder so I'm trying to override that function.

Reply:
Okay, well, you need to write a clear, sentential comment when you use a strange pattern: explain what the pattern does functionally, and why you want to do it.

Secondly, my comments otherwise suggest: we should structure batches to follow what the loss function wants and do away with transposition altogether.